tests/test_request_fingerprinter.py

```diff
@@ -0,0 +1,386 @@
+    "url1,url2,match",
+    (
+        (
+            "https://example.com",
+            "https://example.com",
+            True,
+        ),
+        (
+            "https://example.com",
+            "https://example.com/",
+            True,
+        ),
+        (
+            "https://example.com/a",
+            "https://example.com/b",
+            False,
+        ),
+        (
+            "https://example.com/?1",
+            "https://example.com/?2",
+            False,
+        ),
+        (
+            "https://example.com/?a=1&b=2",
+            "https://example.com/?b=2&a=1",
+            True,
+        ),
+        (
+            "https://example.com",
+            "https://example.com#",
+            True,
+        ),
+        (
+            "https://example.com#",
+            "https://example.com#1",
+            True,
+        ),
+        (
+            "https://example.com#1",
+            "https://example.com#2",
+            True,
+        ),
+    ),
+)
+def test_url(url1, url2, match):
+    crawler = get_crawler()
+    fingerprinter = create_instance(
+        ScrapyZyteAPIRequestFingerprinter, settings=crawler.settings, crawler=crawler
+    )
+    request1 = Request(url1, meta={"zyte_api_automap": True})
+    fingerprint1 = fingerprinter.fingerprint(request1)
+    request2 = Request(url2, meta={"zyte_api_automap": True})
+    fingerprint2 = fingerprinter.fingerprint(request2)
+    if match:
+        assert fingerprint1 == fingerprint2
+    else:
+        assert fingerprint1 != fingerprint2
+
+
+@pytest.mark.parametrize(
+    "params,match",
+    (
+        (
+            {"httpResponseBody": True},
+            True,
+        ),
+        (
+            {"httpResponseBody": True, "httpResponseHeaders": True},
+            True,
+        ),
+        (
+            {"unknownMainOutput": True, "httpResponseBody": True},
+            True,
+        ),
+        (
+            {"httpResponseBody": True, "browserHtml": False},
+            True,
+        ),
+        (
+            {"httpResponseBody": True, "screenshot": False},
+            True,
+        ),
+        (
+            {},
+            False,
+        ),
+        (
+            {"httpResponseBody": False},
+            False,
+        ),
+        (
+            {"httpResponseHeaders": True},
+            False,
+        ),
+        (
+            {"browserHtml": True},
+            False,
+        ),
+        (
+            {"screenshot": True},
+            False,
+        ),
+        (
+            {"browserHtml": True, "screenshot": True},
+            False,
+        ),
+        (
+            {"browserHtml": True, "httpResponseBody": True},
+            False,
+        ),
+        (
+            {"httpResponseBody": True, "screenshot": True},
+            False,
+        ),
+        (
+            {"browserHtml": True, "httpResponseBody": True, "screenshot": True},
+            False,
+        ),
+        (
+            {"unknownMainOutput": True},
```

Comment:
It seems that if unknownMainOutput is browser-related output, this case has a proper behavior, but `{"unknownMainOutput": True, "httpResponseBody": True}` doesn't. And it's the opposite if unknownMainOutput is not a browser-related output: this one is not correct, and the other one is correct.

What do you think about being more conservative, and not taking fragment in account unless there is a known browser-related output set?